src/App.js

```diff
@@ -25,44 +24,98 @@ const generateSquares = () => {
   return squares;
 };
 
+// Creating 2D array
+const reshapSquare = (arr) => {
+  const newArr = [];
+  while (arr.length) newArr.push(arr.splice(0, 3));
+  return newArr;
+};
+
 const App = () => {
-  // This starts state off as a 2D array of JS objects with
-  // empty value and unique ids.
   const [squares, setSquares] = useState(generateSquares());
+  const [nextPlayer, setNextPlayer] = useState(true);
+  const [gameWinner, setWinner] = useState(null);
+
+  const currentPlayer = nextPlayer ? PLAYER_1 : PLAYER_2;
+
+  const changeSquare = (id) => {
+    if (gameWinner !== null) {
+      return;
+    }
+
+    let flattenArray = squares.flat();
+    changedSquares = flattenArray.map((square) => {
+      if (square.id === id && square.value === '') {
+        square.value = currentPlayer;
+        console.log(currentPlayer, 'currentPlayer');
+      }
 
-  // Wave 2
-  // You will need to create a method to change the square
-  //   When it is clicked on.
-  //   Then pass it into the squares as a callback
+      return square;
+    });
+
+    setNextPlayer(!nextPlayer);
+    setSquares(reshapSquare(changedSquares));
+    setWinner(checkForWinner);
+  };
 
   const checkForWinner = () => {
-    // Complete in Wave 3
-    // You will need to:
-    // 1. Go accross each row to see if
-    //    3 squares in the same row match
-    //    i.e. same value
-    // 2. Go down each column to see if
-    //    3 squares in each column match
-    // 3. Go across each diagonal to see if
-    //    all three squares have the same value.
+    let i = 0;
+    while (i < 3) {
```

Comment:
Consider using a for loop since it's a known number of times. A while loop works, but is easier to introduce bugs.